### DISCUSSION TITLE
Make user group fetchable with GraphQL

### DISCUSSION THREAD

**Discussion by Stalex89** [craftcms/cms#12202] [Ideas]
Hello, I would like to have possibility to fetch user group with GraphQL. As an example of use case, I have a user group in Craft called "Manager", and I would like to show user role on the headless frontend (not a twig template) based on the group handle. 

I see that it is possible to query users by group or group ID in GraphQL, but it's not possible to fetch group value.

Could you please add user group to the UserInterface so it can be fetched by GraphQL

Something like this:

```
query getUser {
  user {
    group {
      id
      handle
      slug
    }
  }
}
```

Thank you in advance!

**Comment by vettndr** [comment]
It would be cool to have the user group available in GraphQL!

**Comment by tommysvr** [comment]
This came up in support, so just sharing this as example of a way you could get the group names for a user.


```
Event::on(TypeManager::class, TypeManager::EVENT_DEFINE_GQL_TYPE_FIELDS, function(DefineGqlTypeFieldsEvent $event) {
  if ($event->typeName == 'UserInterface') {
    $event->fields['userGroups'] = [
      'name' => 'userGroupNames',
      'type' => Type::listOf(Type::string()),
      'resolve' => function($source) {
        // As an example; it's likely more performant to the groups up front
        return $source->getGroups();
      }
    ];
  }
});
```

```
query {
  users {
    id
    userGroupNames
  }
}
```
